src/app/ui/palette_listbox.cpp

```diff
@@ -30,17 +28,13 @@ using namespace skin;
 
 class PaletteListItem : public ListItem {
 public:
-  PaletteListItem(PaletteResource* palResource) :
-    ListItem(palResource->name()), m_palResource(palResource) {}
+  PaletteListItem(std::shared_ptr<doc::Palette> palette, const std::string& name) :
+    ListItem(name),
+    m_palette(palette) {}
```

Comment:
nitpicking: it bugs me a bit that the order is palette -> name, but then it becomes name -> palette

Reply:
A base class has to be initialized before the members.
The order a constructor accepts arguments and the order it initializes things are not supposed to be related.